File: server/CasePortal.Api/Models/Membership.cs
namespace CasePortal.Api.Models;

public class Membership
{
    public int       Id                    { get; set; }
    public DateTime  CreatedDate           { get; set; }
    public int       CreatedBy             { get; set; }
    public DateTime  ModifiedDate          { get; set; }
    public int       ModifiedBy            { get; set; }
    public string    Name                  { get; set; } = string.Empty;
    public string    IdType                { get; set; } = "NRIC";
    public string    IdNo                  { get; set; } = string.Empty;
    public string    Insurance             { get; set; } = string.Empty;
    public string    Company               { get; set; } = string.Empty;
    public string    PolicyNo              { get; set; } = string.Empty;
    public decimal   RbEntitlement         { get; set; }
    public decimal   CoPayment             { get; set; }
    public string    CoInsurance           { get; set; } = string.Empty;
    public decimal   Deductible            { get; set; }
    public DateOnly? PolicyEffDate         { get; set; }
    public DateOnly? PolicyExpDate         { get; set; }
    public DateOnly? PolicyLapseDate       { get; set; }
    public string    Status                { get; set; } = "Inforce";
    public string    UnderwritingExclusion { get; set; } = string.Empty;
    public int?      SubmissionId          { get; set; }
}

File: client/src/types/membership.ts
export type Membership = {
  id: string
  date?: string
  name: string
  idType: 'NRIC' | 'Passport'
  idNo: string
  insurance: string
  company: string
  policyNo: string
  rbEntitlement: number
  coPayment: number        // in %
  coInsurance: string      // in % or alphanumeric
  deductible: number       // in currency
  policyEffDate: string
  policyExpDate: string
  policyLapseDate: string
  status: 'Inforce' | 'Expired'
  underwritingExclusion: string
  submissionId?: string
}
